src/components/SearchForm/FuzzySearch.js

```diff
@@ -70,56 +137,7 @@ class FuzzySearch extends PureComponent {
             );
         } else if (reason === 'reset') {
             this.setState({ open: false, value: '' }, () => {
-                if (!value) return;
-                const emoji = value.slice(0, 2);
-                const ident = emoji === emojiMap.INSTRUCTOR ? value.slice(3) : value.slice(3).split(':');
-                const term = RightPaneStore.getFormData().term;
-                resetFormValues();
-                updateFormValue('term', term);
-                switch (emoji) {
-                    case emojiMap.GE_CATEGORY:
-                        updateFormValue(
-                            'ge',
-                            `GE-${ident[0].split(' ')[2].replace('(', '').replace(')', '').toUpperCase()}`
-                        );
-                        break;
-                    case emojiMap.DEPARTMENT:
-                        updateFormValue('deptValue', ident[0]);
-                        updateFormValue('deptLabel', ident.join(':'));
-                        break;
-                    case emojiMap.COURSE:
-                        const deptValue = ident[0].split(' ').slice(0, -1).join(' ');
-                        let deptLabel;
-                        for (const [key, value] of Object.entries(this.state.cache)) {
-                            if (Object.keys(value).includes(deptValue)) {
-                                deptLabel = this.state.cache[key][deptValue].name;
-                                break;
-                            }
-                        }
-                        if (!deptLabel) {
-                            const deptSearch = search(deptValue.toLowerCase());
-                            deptLabel = deptSearch[deptValue].name;
-                            this.setState({
-                                cache: {
-                                    ...this.state.cache,
-                                    [deptValue.toLowerCase()]: deptSearch,
-                                },
-                            });
-                        }
-                        updateFormValue('deptValue', deptValue);
-                        updateFormValue('deptLabel', `${deptValue}: ${deptLabel}`);
-                        updateFormValue('courseNumber', ident[0].split(' ').slice(-1)[0]);
-                        break;
-                    case emojiMap.INSTRUCTOR:
-                        updateFormValue(
-                            'instructor',
-                            Object.keys(this.state.results).filter((x) => this.state.results[x].name === ident)[0]
-                        );
-                        break;
-                    default:
-                        break;
-                }
-                this.props.toggleSearch();
+                this.doSearch(value);
```

Comment:
Making the refactor into a function and adding `enterEvent` separate commits would've made this a bit easier to review cause it's hard to tell if you changed anything after moving it to its own function or just left it the same.

Reply:
Will do in the future, thanks for the advice.